homeassistant/components/aemet/weather.py

```diff
@@ -0,0 +1,120 @@
+"""Support for the AEMET OpenData service."""
+from homeassistant.components.weather import WeatherEntity
+from homeassistant.const import TEMP_CELSIUS
+
+from .const import (
+    ATTR_API_CONDITION,
+    ATTR_API_FORECAST,
+    ATTR_API_HUMIDITY,
+    ATTR_API_PRESSURE,
+    ATTR_API_TEMPERATURE,
+    ATTR_API_WIND_BEARING,
+    ATTR_API_WIND_SPEED,
+    ATTRIBUTION,
+    DOMAIN,
+    ENTRY_NAME,
+    ENTRY_WEATHER_COORDINATOR,
+)
+from .weather_update_coordinator import WeatherUpdateCoordinator
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up AEMET OpenData weather entity based on a config entry."""
+    domain_data = hass.data[DOMAIN][config_entry.entry_id]
+    name = domain_data[ENTRY_NAME]
+    weather_coordinator = domain_data[ENTRY_WEATHER_COORDINATOR]
+
+    unique_id = f"{config_entry.unique_id}"
+    aemet_weather = AemetWeather(name, unique_id, weather_coordinator)
+
+    async_add_entities([aemet_weather], False)
+
+
+class AemetWeather(WeatherEntity):
```

Comment:
Can you inherit from CoordinatorEntity, which might clean up this code a bit?

See https://github.com/home-assistant/core/blob/fd177441b3cfddb22b893b26e91862307bbcc437/homeassistant/helpers/update_coordinator.py#L233

Reply:
This is also inherited from OpenWeatherMap: https://github.com/home-assistant/core/blob/fd177441b3cfddb22b893b26e91862307bbcc437/homeassistant/components/openweathermap/weather.py#L33